src/main/java/com/iexec/core/task/TaskUpdateManager.java

```diff
@@ -278,40 +278,49 @@ void initialized2Running(Task task) {
         }
     }
 
-    public boolean isConsensusReached(Task task) {
+    public boolean isConsensusReached(ReplicatesList replicatesList) {
 
-        Optional<ChainTask> optional = iexecHubService.getChainTask(task.getChainTaskId());
+        Optional<ChainTask> optional = iexecHubService.getChainTask(replicatesList.getChainTaskId());
         if (optional.isEmpty()) return false;
 
         ChainTask chainTask = optional.get();
 
         boolean isChainTaskRevealing = chainTask.getStatus().equals(ChainTaskStatus.REVEALING);
 
         int onChainWinners = chainTask.getWinnerCounter();
-        int offChainWinners = isChainTaskRevealing ? replicatesService.getNbValidContributedWinners(task.getChainTaskId(), chainTask.getConsensusValue()) : 0;
+        int offChainWinners = isChainTaskRevealing
+                ? replicatesService.getNbValidContributedWinners(replicatesList.getReplicates(), chainTask.getConsensusValue())
+                : 0;
         boolean offChainWinnersGreaterOrEqualsOnChainWinners = offChainWinners >= onChainWinners;
 
         return isChainTaskRevealing && offChainWinnersGreaterOrEqualsOnChainWinners;
     }
 
     void running2ConsensusReached(Task task) {
         boolean isTaskInRunningStatus = task.getCurrentStatus().equals(RUNNING);
-        boolean isConsensusReached = isConsensusReached(task);
+        final String chainTaskId = task.getChainTaskId();
+        final Optional<ReplicatesList> oReplicatesList = replicatesService.getReplicatesList(chainTaskId);
```

Comment:
Wouldn't this result in the same synchronization problems ? The task and the replicates list hasn't been fetched at the same time. I think the replicateList should be fetched from the db by the caller at the same time as the task and passed to this method.
This makes me think about a `MetaTask` object that contains at the same time, the task details and the replicatesList. That way we guarantee that the task and it's replicate are always synchronized.

Reply:
I don't think there could be the same sync problem here:
- a same task can't be updated twice at the same time so its status won't change between its retrieval and the start of `running2ConsensusReached`;
- replicates can be updated between the task retrieval and the start of `running2ConsensusReached` but I don't think it matters. The "worst" case would be an update of one or multiple replicates in that period, but that would simply lead to another result for the method... Which would have been computed later otherwise.

Reply:
Getting back to this: while I still agree with what I said in my last comment, having a `MetaTask` object could be great. It'd be computed at the start of `TaskUpdateManager#updateTask` and passed down to all `status12status2` methods in `TaskUpdateManager`. That'd help ensuring there's no desynchronization when we are for example in this block:
```
case RUNNING:
  running2ConsensusReached(task);
  running2RunningFailed(task);
  initializedOrRunning2ContributionTimeout(task);
  break;
```
This would go with the other optimizations I mentioned [there](https://github.com/iExecBlockchainComputing/iexec-core/pull/490#discussion_r786713900).

I'd like to work on this point once https://github.com/iExecBlockchainComputing/iexec-core/pull/484 is merged.